online-shop/src/app/components/product-list/product-list.component.ts

```diff
@@ -1,22 +1,29 @@
 import { Component, OnInit } from '@angular/core';
 import { headers } from '../../mockup/headers';
-import { HttpClient } from '@angular/common/http';
-import { Product } from 'src/app/interfaces/productInterface';
+import { LoginService } from 'src/app/services/login.service';
+import { AppState } from 'src/app/store/state/app.state';
+import { select, Store } from '@ngrx/store';
+import { selectIsLoading, selectProductList } from 'src/app/store/selectors/product.selectors';
+import { GetAllProducts } from 'src/app/store/actions/product.actions';
+import { take } from 'rxjs/operators';
 
 @Component({
   selector: 'app-product-list',
   templateUrl: './product-list.component.html',
   styleUrls: ['./product-list.component.css'],
 })
 export class ProductListComponent implements OnInit {
-  products: Product[] = [];
+  products: any;
```

Comment:
Try not to use the type **any**, I know it might be the best option when you don't know the type of an object, but it's actually a bad choice to do so. This type cannot guarantee you against null or undefined checks. Moreover, for another programmer that has to interpret your code and come with more features, it will be hard to understand the attributes of such an object. This could even introduce bugs if the object with type **any** is used in many places and handles complex logic. 

In your case the type of products should be **Observable<Product[]>** if I am not mistakes, but please check it.